stellar-multitenant/pkg/internal/provisioning/manager.go

```diff
@@ -49,238 +59,261 @@ func rollbackTenantDataSetupErrors() []error {
-			return nil, fmt.Errorf("%w: %w", ErrTenantSchemaFailed, tenantSchemaFailedErr)
-		}
+	// Last step when no errors - fund tenant distribution account
+	fundErr := m.fundTenantDistributionAccount(ctx, *t.DistributionAccount)
+	if fundErr != nil {
+		// error already wrapped
+		return nil, fundErr
+	}
 
-		// Connecting to the tenant database schema
-		tenantSchemaConnectionPool, err := db.OpenDBConnectionPool(u)
-		if err != nil {
-			return nil, fmt.Errorf("%w: opening database connection on tenant schema: %w", ErrOpenTenantSchemaDBConnectionFailed, err)
-		}
-		defer tenantSchemaConnectionPool.Close()
-
-		tenantDataSetupErr := func() error {
-			err = services.SetupAssetsForProperNetwork(ctx, tenantSchemaConnectionPool, utils.NetworkType(networkType), services.DefaultAssetsNetworkMap)
-			if err != nil {
-				return fmt.Errorf("running setup assets for proper network: %w", err)
-			}
-
-			err = services.SetupWalletsForProperNetwork(ctx, tenantSchemaConnectionPool, utils.NetworkType(networkType), services.DefaultWalletsNetworkMap)
-			if err != nil {
-				return fmt.Errorf("running setup wallets for proper network: %w", err)
-			}
-
-			// Updating organization's name
-			models, getTntModelsErr := data.NewModels(tenantSchemaConnectionPool)
-			if getTntModelsErr != nil {
-				return fmt.Errorf("getting models: %w", err)
-			}
-
-			err = models.Organizations.Update(ctx, &data.OrganizationUpdate{Name: organizationName})
-			if err != nil {
-				return fmt.Errorf("updating organization's name: %w", err)
-			}
-
-			// Creating new user and sending invitation email
-			authManager := auth.NewAuthManager(
-				auth.WithDefaultAuthenticatorOption(tenantSchemaConnectionPool, auth.NewDefaultPasswordEncrypter(), 0),
-			)
-			s := services.NewCreateUserService(models, tenantSchemaConnectionPool, authManager, m.messengerClient)
-			_, err = s.CreateUser(ctx, auth.User{
-				FirstName: userFirstName,
-				LastName:  userLastName,
-				Email:     userEmail,
-				IsOwner:   true,
-				Roles:     []string{"owner"},
-			}, uiBaseURL)
-			if err != nil {
-				return fmt.Errorf("creating user: %w", err)
-			}
-
-			return nil
-		}()
-		if tenantDataSetupErr != nil {
-			return nil, fmt.Errorf("%w: %w", ErrTenantDataSetupFailed, tenantDataSetupErr)
-		}
+	return t, nil
+}
 
-		// Provision distribution account for tenant if necessary
-		distributionAccPubKeys, err := m.SubmitterEngine.DistAccountSigner.BatchInsert(ctx, 1)
-		if err != nil {
-			if errors.Is(err, signing.ErrUnsupportedCommand) {
-				log.Ctx(ctx).Warnf(
-					"Account provisioning not needed for distribution account signature client type %s: %v",
-					m.SubmitterEngine.DistAccountSigner.Type(), err)
-			} else {
-				return nil, fmt.Errorf("%w: provisioning distribution account: %w", ErrProvisionTenantDistributionAccountFailed, err)
-			}
-		}
+func (m *Manager) handleProvisioningError(ctx context.Context, name string, err error, t *tenant.Tenant) error {
```

Comment:
Do we need the name here as a parameter if we also have the tenant ?

Reply:
nope, thanks for calling it out